### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by soniaroh**
- created a join table
- wrote some tests
- transferred image source to S3
- finished carousel format and added CSS

### CODE REVIEW

File: client/src/PzElement.jsx [hunk 1]
```diff
@@ -1,14 +1,65 @@
...
-		// temporarily using inline sizing to test
-		<h3>
-			<img width="160" height="160" src={props.element.imageURL} />
-			{props.element.productName}
-			{props.element.price}
-		</h3>
-	);
+// function creates star ratings based on the number of ratings, decimals are not considered yet
+const starMaker = function (num) {
+  const n = 5 - num;
```
Comment (michchen58) [inline client/src/PzElement.jsx]: It'd be helpful to have more descriptive variables names

File: client/src/PzSlider.jsx [hunk 2]
```diff
@@ -1,33 +1,23 @@
...
-					))}
-				</Slider>
-			</div>
-		</div>
-	);
+const PzSlider = (props) => {
+  return (
+    <div>
+     
+      <h3 className={styles.title}>Customers who bought this item also bought</h3>
```
Comment (michchen58) [inline client/src/PzSlider.jsx]: Use destructuring assignment (airbnb style)

File: .env [hunk 3]
```diff
@@ -1,3 +1,3 @@
 DB_HOST=localhost
 DB_USER=root
```
Comment (avincenthill) [inline .env:2]: gitignore this please

File: client/src/PzElement.jsx [hunk 4]
```diff
@@ -1,14 +1,65 @@
...
-	return (
-		// temporarily using inline sizing to test
-		<h3>
-			<img width="160" height="160" src={props.element.imageURL} />
-			{props.element.productName}
-			{props.element.price}
-		</h3>
-	);
+// function creates star ratings based on the number of ratings, decimals are not considered yet
+const starMaker = function (num) {
```
Comment (avincenthill) [inline client/src/PzElement.jsx]: concur with Michelle about more descriptive input here

File: client/src/PzElement.jsx [hunk 5]
```diff
@@ -1,14 +1,65 @@
...
+  const n = 5 - num;
+  const lit = [];
+  const dim = [];
+  for (var i = 0; i < num; i++) {
+    lit.push(
+			<img width="18" height="18" src="https://files.slack.com/files-pri/TBV9WH1HR-FCPN3R8TV/fullstar.png" />
```
Comment (avincenthill) [inline client/src/PzElement.jsx]: rm indent here

File: client/src/SlideShow.jsx [hunk 6]
```diff
@@ -1,42 +1,108 @@
...
+  }
+
+  componentDidMount() {
+    fetch('http://localhost:4043/products')
+      .then(response => response.json())
+      .then(({ data }) => {
+        const arr = [];
+        let numOfItem = 6;
+        if (data.length < 6) { numOfItem = data.length; }
+        for (let i = 0; i < numOfItem; i++) {
```
Comment (michchen58) [inline client/src/SlideShow.jsx]: maybe use forEach

File: client/src/PzSlider.jsx [hunk 7]
```diff
@@ -1,33 +1,23 @@
...
-
-	const settings = {
-		dots: true,
-		infinite: true,
-		speed: 500,
-		slidesToShow: 6,
-		slidesToScroll: 5
-	};
-	return (
-		<div>
```
Comment (avincenthill) [inline client/src/PzSlider.jsx:16]: too much indentation  - use prettier to autoformat here

File: client/src/SlideShow.jsx [hunk 8]
```diff
@@ -1,42 +1,108 @@
...
+    });
+  }
+
+
+  render() {
+    return (
+      <div>
+        <span className={styles.page}>
+        Page
+          {' '}
```
Comment (michchen58) [inline client/src/SlideShow.jsx]: try backtick referencing

File: client/src/SlideShow.jsx [hunk 9]
```diff
@@ -1,42 +1,108 @@
...
+    super(props);
+    this.state = {
+      products: [],
+      pageNum: 0,
+      pageProducts: [],
+    };
+  }
+
+  componentDidMount() {
+    fetch('http://localhost:4043/products')
```
Comment (avincenthill) [inline client/src/SlideShow.jsx]: use env vars here, idk how fetch works but this line will change in deployment

File: database/index.js [hunk 10]
```diff
@@ -1,68 +1,77 @@
...
-        } else {
-            callback(null, data)
-        }
-    })
-}
-
-// temporary function for testing purposes
+const getRelatedItems = () => {
+  for (let i = 1; i <= 100; i++) {
+    const relatedItems = ((Math.random() * 38) + 6);
```
Comment (michchen58) [inline database/index.js:51]: where did 38 come from?

File: client/src/main.css [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+    
+    overflow: hidden;
+    position: absolute;
+   
+   right: 0%;
+   left:3%;
+}
+
+:local(.text) {
+    font-family: "Amazon Ember", Arial, sans-serif;
```
Comment (avincenthill) [inline client/src/main.css:16]: interesting

**Review by michchen58 — APPROVED** [review]

File: jestsetup.js [hunk 12]
```diff
@@ -0,0 +1,68 @@
+
+import Enzyme, { shallow, render, mount } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+
+// React 16 Enzyme adapter
+Enzyme.configure({ adapter: new Adapter() });
+// Make Enzyme functions available in all test files without importing
+global.shallow = shallow;
```
Comment (avincenthill) [inline jestsetup.js]: check if this is best practice

File: test/client.test.js [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+import Enzyme, { shallow, render, mount } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import App from '../client/src/App';
+import RelatedItems from '../client/src/SlideShow';
+import PzSlider from '../client/src/PzSlider';
+
+// React 16 Enzyme adapter
+Enzyme.configure({ adapter: new Adapter() });
+// Make Enzyme functions available in all test files without importing
+global.shallow = shallow;
```
Comment (avincenthill) [inline test/client.test.js:11]: do you need to do this twice if you're altering global?

File: test/client.test.js [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+  test('RelatedItems renders a document title', () => {
+    const wrapper = shallow(
+    <PzSlider items={this.state.pageProducts} />,
+    );
+    expect(wrapper.prop('item')).toEqual('{this.state.pageProducts}');
+  });
+
+  // testing event handlers
+
+  test('pass a selected value to the onClick handler', () => {
```
Comment (avincenthill) [inline test/client.test.js:55]: cool test

**Review by avincenthill — CHANGES_REQUESTED** [review]

File: database/index.js (hunk 10 above)
Comment (soniaroh) [reply database/index.js:51]: 6 is the min number of items I wanted to have on each page before they run out, and 38 is just a random number I chose to generate random numbers between two points for related items.

**Pull request closed without merging**